microcosm_postgres/factories.py

```diff
@@ -57,14 +63,24 @@ def configure_sqlalchemy_engine(graph):
         database_name,
     )
 
-    return create_engine(
-        uri,
+    connection_args = dict(
         pool_size=graph.config.postgres.pool_size,
         pool_timeout=graph.config.postgres.pool_timeout,
         max_overflow=graph.config.postgres.max_overflow,
         echo=bool(graph.config.postgres.echo),
+        connect_args=dict(
+            sslmode="require" if graph.config.postgres.require_ssl else "prefer"
+        )
     )
 
+    if graph.config.postgres.verify_ssl:
+        connection_args["connect_args"] = {
+            "sslmode": "verify-full",
```

Comment:
This sneakily overrides the "prefer" if verify_ssl is on but require_ssl is off. If we think require off/verify on is an invalid configuration, we should explicitly check for it and call it out, not do sneaky things...

Reply:
You are correct - I made assumptions based on common sense, but can refactor to be more explicit.